plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationProviderImpl.kt

```diff
@@ -39,6 +41,9 @@ internal class LocationProviderImpl(context: Context, var settings: LocationComp
   private lateinit var runnable: Runnable
   private var updateDelay = INIT_UPDATE_DELAY
 
+  private val locationEngineCallback: LocationEngineCallback<LocationEngineResult> =
```

Comment:
Since this variable is member of the class - it seem to still hold an implicit reference to the enclosing object (LocationProviderImpl).

Reply:
`LocationProviderImpl` holds the reference to `LocationEngineCallback`. The change in this pr is to avoid `LocationEngineCallback` holds the reference to `LocationProviderImpl`

Reply:
Ah indeed, seems I've confused with inner classes somehow.